ui/pages/onboarding-flow/create-password/create-password.js

```diff
@@ -219,123 +143,89 @@ export default function CreatePassword({
-              <ButtonLink
-                variant={TextVariant.bodySm}
-                data-testid="show-password"
-                className="create-password__form--password-button"
-                onClick={(e) => {
-                  e.preventDefault();
-                  setShowPassword(!showPassword);
-                }}
-                marginBottom={1}
-                // This type="a" prop is needed so that the button doesn't submit the form
-                // or keep the "Show/Hide" alive when clicked outside of the button
-                type="a"
-                href="#"
-              >
-                {showPassword ? t('hide') : t('show')}
-              </ButtonLink>
-            }
+          width={BlockSize.Full}
+        >
+          <ButtonIcon
+            iconName={IconName.ArrowLeft}
+            color={IconColor.iconDefault}
+            size={ButtonIconSize.Md}
+            data-testid="create-password-back-button"
+            type="button"
+            onClick={() => history.goBack()}
+            ariaLabel="back"
           />
-          <FormField
-            dataTestId="create-password-confirm"
-            marginTop={3}
-            onChange={handleConfirmPasswordChange}
-            password={!showPassword}
-            error={confirmPasswordError}
-            titleText={t('confirmPassword')}
-            value={confirmPassword}
-            titleDetail={
-              isValid && (
-                <div className="create-password__form--checkmark">
-                  <Icon name={IconName.Check} />
-                </div>
-              )
+        </Box>
+        <Box
+          justifyContent={JustifyContent.flexStart}
+          marginBottom={4}
+          width={BlockSize.Full}
+        >
+          <Text variant={TextVariant.bodyMd} color={TextColor.textAlternative}>
+            {t('stepOf', [
+              firstTimeFlowType === FirstTimeFlowType.import ? 2 : 1,
+              firstTimeFlowType === FirstTimeFlowType.import ? 2 : 3,
+            ])}
+          </Text>
+          <Text variant={TextVariant.headingLg} as="h2">
+            {t('createPassword')}
+          </Text>
+        </Box>
+        <PasswordForm onChange={(newPassword) => setPassword(newPassword)} />
+      </Box>
+      <Box>
+        <Box
+          className="create-password__terms-container"
+          alignItems={AlignItems.center}
+          justifyContent={JustifyContent.spaceBetween}
+          marginBottom={4}
+        >
+          <Checkbox
+            inputProps={{ 'data-testid': 'create-password-terms' }}
+            alignItems={AlignItems.flexStart}
+            isChecked={termsChecked}
+            onChange={() => {
+              setTermsChecked(!termsChecked);
+            }}
+            label={
+              <Text variant={TextVariant.bodySm} marginLeft={1}>
+                {t('passwordTermsWarning')}
+                &nbsp;
+                {createPasswordLink}
+              </Text>
             }
```

Comment:
non-blocking: another way of doing this and reducing nesting would be 

```suggestion
            label={
              <>
                {t('passwordTermsWarning')}
                &nbsp;
                {createPasswordLink}
              </>
            }
            labelProps={{
              variant: TextVariant.bodySm,
              marginLeft: 1,
            }}
```